kola/tests/docker/torcx_docker_flag.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"regexp"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform/conf"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:         dockerTorcxFlagFile,
+		ClusterSize: 1,
+		Name:        "docker.torcx-flag-file",
+		MinVersion:  semver.Version{Major: 1548},
+		UserData: conf.ContainerLinuxConfig(`
+storage:
+  files:
+    - filesystem: root
+      path: /etc/coreos/docker-1.12
+      contents:
+        inline: yes
+`),
+	})
+}
+
+func dockerTorcxFlagFile(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	checkVersions := func(expectedRefRE, expectedVerRE string) {
+		ref := getTorcxDockerReference(c, m)
+		if !regexp.MustCompile(expectedRefRE).MatchString(ref) {
+			c.Errorf("reference %s did not match %q", ref, expectedRefRE)
+		}
+
+		ver := getDockerServerVersion(c, m)
+		if !regexp.MustCompile(expectedVerRE).MatchString(ver) {
+			c.Errorf("version %s did not match %q", ver, expectedVerRE)
+		}
+	}
+
+	// flag=yes
+	checkVersions(`^1\.12$`, `^1\.12\.`)
+
+	// flag=no
+	if _, err := m.SSH("echo no | sudo tee /etc/coreos/docker-1.12"); err != nil {
+		c.Fatalf("couldn't set docker flag to no: %s", err)
+	}
+	if err := m.Reboot(); err != nil {
+		c.Fatalf("could not reboot: %v", err)
+	}
+	if _, err := m.SSH(`sudo rm -rf /var/lib/docker`); err != nil {
```

Comment:
since we're only upgrading, this shouldn't be necessary.

Reply:
It shouldn't hurt, though.  Otherwise it would still detect the `overlay` driver from starting at 1.12, which would fail the driver test (if it ran).

Reply:
Yeah, no preference. I guess since we didn't run a container it won't even take seconds to delete.

This LGTM as well.